testing/selenium_tests.py

```diff
@@ -0,0 +1,625 @@
+        """Is there any data?"""
+        panel = load_panel("MITRE ATT&CK Technique")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_signals_details(self):
+        """Is there any data?"""
+        panel = load_panel("Signals Details")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_full_event_logs(self):
+        """Is there any data?"""
+        panel = load_panel("Full Event Logs")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+class HealthCheckTests(unittest.TestCase):
+    """Test cases for the HealthCheck Dashboard"""
+    #2/6/2024, main branch on lme. The health check dashboard has an odd dashboard menu. This will likely need updating.
+
+    def setUp(self):
+        # The dashboard ID is hard-coded in the ndjson file
+        dashboard_id = "51fe1470-fa59-11e9-bf25-8f92ffa3e3ec"
+        driver.get(f"https://ls1/app/dashboards#/view/{dashboard_id}")
+        expected_cond = EC.presence_of_element_located((By.CLASS_NAME, "react-grid-layout"))
+        WebDriverWait(driver, args.timeout).until(expected_cond)
+
+    def test_total_hosts(self):
+        """Is there any data?"""
+        panel = load_panel("Alpha - Health Check - Total Hosts")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_users_seen(self):
+        """Is there any data?"""
+        panel = load_panel("Users seen")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_number_of_admins(self):
+        """Is there any data?"""
+        panel = load_panel("Alpha - Health Check - Number of Admins - Metric (converted)")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_events_by_machine(self):
+        """Is there any data?"""
+        panel = load_panel("Events by machine")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+    def test_unexpected_shutdowns(self):
+        """Is there any data?"""
+        panel = load_panel("Unexpected shutdowns")
+        self.assertFalse("No results found" in panel.get_attribute("innerHTML"))
+
+options = webdriver.ChromeOptions()
+# options.add_argument("--headless=new")
+options.add_experimental_option("detach", True) # Make browser stay open, in reality we'd
+# probably prefer headless
+
+s = Service(ChromeDriverManager().install())
+driver = webdriver.Chrome(service=s, options=options)
+
+try:
+    login(os.environ['ELASTIC_PASSWORD'])
+except KeyError:
+    MESSAGE = "Error: Elastic password not set. Should be saved as env variable, ELASTIC_PASSWORD."
+    print(MESSAGE, file=sys.stderr)
+    sys.exit(1)
+unit_argv = [sys.argv[0]] + unittestArgs
+unittest.main(argv=unit_argv)
+
+# driver.stop_client() # uncommented so the browser window stays open for debugging
+# driver.close()
+# driver.quit()

```

Comment:
Also relevant code for headless/detached. Do we want the browser window to stay open? Or do we want the tests to clean up after itself? Maybe we'd want a switch for this too.

Reply:
Added a switch for this